easybuild/easyblocks/i/imkl.py

```diff
@@ -261,14 +262,21 @@ def build_mkl_fftw_interfaces(self, libdir):
                 tmpbuild = tempfile.mkdtemp(dir=self.builddir)
                 self.log.debug("Created temporary directory %s" % tmpbuild)
 
+                # Avoid unused command line arguments (-Wl,rpath...) causeing errors when using RPATH
+                # See https://github.com/easybuilders/easybuild-easyconfigs/pull/18439#issuecomment-1662671054
+                if build_option('rpath'):
+                    cflags = flags + ' -Wno-unused-command-line-argument'
```

Comment:
This option is probably only supported for Intel oneAPI (Clang-based) compilers, so this needs an extra condition?

Reply:
Had some discussion on Slack. This is correct: it's supported by Clang-based compilers. Hence we should limit it to those. The problem that I'm trying to address in this PR should also only occur for Clang based compilers (that's why we haven't run into it with prior versions if imkl-fftw: we didn't use the oneAPI compilers yet by default there).